pkg/util/usage/usage.go

```diff
@@ -12,18 +12,18 @@ import (
 	"github.com/grafana/dskit/flagext"
 
 	"github.com/grafana/mimir/pkg/ingester/activeseries"
-	"github.com/grafana/mimir/pkg/mimir"
 	"github.com/grafana/mimir/pkg/util/fieldcategory"
 )
 
-// usage prints command-line usage. If mf.printHelpAll is false then only basic flags are included, otherwise all flags are included.
-func usage(mf *mainFlags, cfg *mimir.Config) error {
+// Usage prints command-line usage.
+// printAll controls whether only basic flags or all flags are included.
+// configs are expected to be pointers to structs.
+func Usage(printAll bool, configs ...interface{}) error {
```

Comment:
optional: I find it easier to understand methods at a glance that use more explicit arguments instead of bools. For example, this method could accept a slice (or set) of field categories that it should generate help for instead of a "show all" flag. You could make this less verbose in the common cases by defining slices of `fieldcategory.All` and `fieldcategory.BasicAndExperimental` or similar.

Reply:
`fieldCategory` defines an enum for the category values, so we could pass a "max category level" to be more explicit. I'll take a look at that in a follow-up PR.